sleap_roots/convhull.py

```diff
@@ -1 +1,56 @@
 """Convex hull fitting and derived trait calculation."""
+
+import numpy as np
+from scipy.spatial import ConvexHull, convex_hull_plot_2d
+from scipy.spatial.distance import pdist
+from typing import Tuple
+
+
+def get_convhull(
```

Comment:
This is maybe a semantic preference, but I'd call this `get_convhull_features`.

The convex hull is technically defined by a set of **points**, so that's what I'd expect to get back from calling a function named `get_convhull`.

A better organization here might be to split this into two functions, where one just gets the convhull points:

```py
def get_convhull(pts: np.ndarray) -> ConvexHull:
    pts = pts.reshape(-1, 2)
    pts = pts[~(np.isnan(pts).any(axis=-1))]

    if len(pts) <= 2:
        return np.full((7,), np.nan)

    # Get Convex Hull
    hull = ConvexHull(pts)

    return hull
```

And the second one which takes the `ConvexHull` instance and gives you back features. I appreciate it's more cumbersome to have to call two functions, so you could add a convenience checker to compute the convhull if needed:

```py
def get_convhull_features(pts: Union[np.ndarray, ConvexHull]):
    hull = pts if type(pts) == ConvexHull else get_convhull(pts)
    
    # perimeter
    perimeter = hull.area
    ...
```